### PR TITLE
Fix error logging logic with interceptors

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

> $Subject

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5877

- The unhandled internal errors such as dispatching errors, data-binding errors, data-validation errors and transport level errors will be printed at the native level

```console
error: no header value found for 'foo'

error: header binding failed for parameter: 'id'
cause: error occurred while converting "hello" to the target type

error: no matching resource found for path : /api/path, method : GET
```
- The other unhandled errors returned from the resource will be logged using `ballerina/log:printError` method at the default error interceptor

```console
time=2024-01-29T10:18:11.872+05:30 level=ERROR module=ballerina/http message="unhandled error
 returned from the service" error={"causes":[],"message":"Simulated error","detail":{},"stackTrace":
[{"callableName":"getError","moduleName":"ballerina.http_misc_tests$test.2.tests.http_resource_
function_return_test","fileName":"tests/http_resource_function_return_test.bal","lineNumber":54},
{"callableName":"$default$checkErrorReturn","moduleName":"ballerina.http_misc_tests$test.2.
$anonType$_51","fileName":"tests/http_resource_function_return_test.bal","lineNumber":38}]} 
path="/checkErrorReturn" method="GET"
```

- The above applies to status code errors as well

```console
time=2024-01-29T10:15:08.976+05:30 level=ERROR module=ballerina/http message="unhandled error
 returned from the service" error={"causes":[],"message":"Album not found","detail":{"body":{
"timeStamp":"2024-01-29T04:45:08.939823Z","message":"Album not found for title: test"}},
"stackTrace":[{"callableName":"$get$albums$^","moduleName":"tharmigan.intercept_error.0.$anonType$_0",
"fileName":"main.bal","lineNumber":37}]} path="/albums/test" method="GET"
```

## Examples

N/A

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests (Testing is done manually as the fix is related to logs)
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/http/api/ResourceDataElement.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ private Resource validateHTTPMethod(List<Resource> resources, HttpCarbonMessage
         }
         if (!isOptionsRequest) {
             String message = "Method not allowed";
```
Comment (dilanSachi) [inline native/src/main/java/io/ballerina/stdlib/http/api/ResourceDataElement.java:139]: creating a new variable is redundant right?

File: native/src/main/java/io/ballerina/stdlib/http/api/service/signature/AllHeaderParams.java [hunk 2]
```diff
@@ -158,7 +158,7 @@ private BMap<BString, Object> processHeaderRecord(HeaderParam headerParam, HttpH
                     return null;
                 } else {
                     String message = "no header value found for '" + key + "'";
```
Comment (dilanSachi) [inline native/src/main/java/io/ballerina/stdlib/http/api/service/signature/AllHeaderParams.java:160]: i guess we can make a constant string and use `String.format` in the used places

Comment (TharmiganK) [reply native/src/main/java/io/ballerina/stdlib/http/api/service/signature/AllHeaderParams.java]: Fixed via [77f7547](https://github.com/ballerina-platform/module-ballerina-http/pull/1846/commits/77f754789448ea6acaeb14b28a2d42d42e072a68)

**Review by dilanSachi — APPROVED** [review]
LGTM

**Pull request merged**